lib/ortho_functions.py

```diff
@@ -238,9 +204,381 @@
+                    gtf = ds.GetGeoTransform()
+                    ulx = gtf[0] + 0 * gtf[1] + 0 * gtf[2]
+                    uly = gtf[3] + 0 * gtf[4] + 0 * gtf[5]
+                    urx = gtf[0] + xsize * gtf[1] + 0 * gtf[2]
+                    ury = gtf[3] + xsize * gtf[4] + 0 * gtf[5]
+                    llx = gtf[0] + 0 * gtf[1] + ysize * gtf[2]
+                    lly = gtf[3] + 0 * gtf[4] + ysize * gtf[5]
+                    lrx = gtf[0] + xsize * gtf[1] + ysize * gtf[2]
+                    lry = gtf[3] + xsize * gtf[4] + ysize * gtf[5]
+                ds = None
+
+                ####  Create geometry objects
+                ul_geom = ogr.Geometry(ogr.wkbPoint)
+                ul_geom.AddPoint(ulx, uly)
+                ur_geom = ogr.Geometry(ogr.wkbPoint)
+                ur_geom.AddPoint(urx, ury)
+                lr_geom = ogr.Geometry(ogr.wkbPoint)
+                lr_geom.AddPoint(lrx, lry)
+                ll_geom = ogr.Geometry(ogr.wkbPoint)
+                ll_geom.AddPoint(llx, lly)
+
+                ring = ogr.Geometry(ogr.wkbLinearRing)
+                ring.AddPoint(ulx, uly)
+                ring.AddPoint(urx, ury)
+                ring.AddPoint(lrx, lry)
+                ring.AddPoint(llx, lly)
+                ring.AddPoint(ulx, uly)
+                image_geom = ogr.Geometry(ogr.wkbPolygon)
+                image_geom.AddGeometry(ring)
+                g_srs = srs_wgs84
+
+                #### Create source srs objects
+                s_srs = utils.osr_srs_preserve_axis_order(osr.SpatialReference(proj))
+                try:
+                    sg_ct = osr.CoordinateTransformation(s_srs, g_srs)
+                except RuntimeError as e:
+                    logger.error(f"Source image coordinate system error: {self.src_image} - {e}")
+                    rc = 1
+                else:
+                    #### Transform geometries to geographic
+                    if not s_srs.IsSame(g_srs):
+                        ul_geom.Transform(sg_ct)
+                        ur_geom.Transform(sg_ct)
+                        ll_geom.Transform(sg_ct)
+                        lr_geom.Transform(sg_ct)
+                        image_geom.Transform(sg_ct)
+                    logger.info("Geographic extent: %s", str(image_geom))
+
+                    #### Get geographic Envelope
+                    self.minlon, self.maxlon, self.minlat, self.maxlat = image_geom.GetEnvelope()
+
+                    ## Determine output image projection if applicable
+                    if type(args.epsg) is str:
+                        self.cent_lat = (self.minlat + self.maxlat) / 2
+                        self.cent_lon = (self.minlon + self.maxlon) / 2
+                        self.epsg = get_epsg_from_lat_lon(self.cent_lat, self.cent_lon, mode=args.epsg, utm_nad83=args.epsg_utm_nad83)
+                        logger.info("Automatically selected output projection EPSG code: %d", self.epsg)
```

Comment:
Log fires on every scene. Consider changing to `logger.debug()` for opt-in verbosity.